core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -264,16 +239,95 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
     authorizerLogger.debug(s"Principal = $principal is $permissionType Operation = $operation from host = $host on resource = $resource")
   }
 
+  /**
+    * Safely updates the resources acls by ensureing reads and writes respect the expected zookeeper version.
+    * Continues to retry until it succesfully updates.
+    *
+    * @param resource the resource to change acls for
+    * @param getNewAcls function to transform existing acls to new acls
+    * @return boolean indicating if a change was made
+    */
+  private def updateResourceAcls(resource: Resource)(getNewAcls: Set[Acl] => Set[Acl]): Boolean = {
+    val path = toResourcePath(resource)
+
+    var currentVersionedAcls =
+      if(aclCache.contains(resource))
+        getAclsFromCache(resource)
+      else
+      getAclsFromZk(resource)
+    var newVersionedAcls = currentVersionedAcls
+    var writeComplete = false
+    while (!writeComplete) {
```

Comment:
The problem with sync is that it is an asynchronous call and I believe ZkClient only supports the synchronous API. I don't think it is worth using the sync call here because it won't stop concurrent accesses anyway. When we move to the async API for the controller, then perhaps we can make this change here.